### PR TITLE
fix(worktree): a linked node_modules is a symlink on POSIX

### PR DESCRIPTION
**Pull request by rimusz**
## What this fixes

`scripts/worktree.test.js` has been failing on `main` since 2c7199b1, so **every PR inherits a red Backend (vitest) check**. Main's own run [33596904066](https://github.com/agnt-gg/agnt/actions/runs/33596904066) (@ acc865d1) fails the same two tests:

- `detaches links first, removes the worktree, deletes the merged branch`
- `keeps an unmerged branch rather than -D it on your behalf`

This branch is based on `acc865d1` and touches nothing else.

## Root cause — one bug, two symptoms

`git worktree` links are **junctions on Windows** but **symlinks on macOS/Linux**, and git classifies the two differently: a junction is a *directory*, a symlink is a *file*. Every rule that named these links assumed the Windows spelling.

Demonstrated with a two-case probe against the pre-fix `.gitignore`:

| `node_modules` is… | `git status --porcelain -uall` |
|---|---|
| a real directory | *(ignored)* |
| a symlink | `?? node_modules` |

**1. `.gitignore` said `node_modules/`.** A trailing slash matches directories only, so the link was never ignored. It shows up as untracked in every worktree — and `git add .` inside one **commits a symlink pointing at the primary checkout**, the exact accident the pattern exists to prevent.

**2. `removeWorktree`'s dirty filter tested `file.startsWith('node_modules/')`**, which cannot match the bare `node_modules` that POSIX git reports. So `wt remove` declared uncommitted changes and refused every worktree it had just created.

The second test failure is downstream of the first: because the links aren't ignored, `git add .` commits them; `detachLinks` then deletes those tracked paths, leaving staged deletions, so `git worktree remove` refuses. Each step of that chain was reproduced directly.

## The change

- **`.gitignore`** — dropped the trailing slash on the two paths `worktree.mjs` actually links (`node_modules`, `frontend/node_modules`). Other entries are untouched.
- **`scripts/worktree.mjs`** — extracted the filter into a pure exported `isLinkNoise()` covering both spellings (`f === d || f.startsWith(d + '/')`), so it is testable without a filesystem.
- **`scripts/worktree.test.js`** — 13 → 18 tests.

The new `.gitignore` contract test copies the **real** `.gitignore` into a probe repo and asserts a symlinked `node_modules` is not reported. It deliberately does not assert against the fixture's own copy: a fixture that merely *claims* to mirror the repo cannot catch the repo drifting.

## Verification

- `scripts/worktree.test.js` — **18/18**, and green inside the full run
- Full `npm test` — **5508 passed**. The only failures are `browserFallbackSurface.test.js` (*"Brave does not appear to be installed on this machine"*), which fail identically on an unmodified checkout and pass in CI
- `npm run eol:check` — clean (2621 files, would rewrite 0)

**Mutation-tested**, each mutation killed by a different test:

| Mutation | Killed by |
|---|---|
| restore the trailing slashes | `.gitignore` contract test |
| drop the bare-entry match | POSIX `isLinkNoise` test |
| naive `startsWith(d)` prefix match | *never swallows the user's own work* |

That middle one is why the pure function was extracted: dropping the bare-entry match does **not** break the integration tests once `.gitignore` is fixed, so without a unit test that layer would sit unguarded.

## Note for reviewers on Windows

This is POSIX-only in effect. On Windows the links are junctions, git reports them as directories, and both the old and new patterns match — so behaviour there is unchanged. The `isLinkNoise` tests cover both spellings explicitly so neither platform can regress silently.

### CODE REVIEW

File: scripts/worktree.mjs [hunk 1]
```diff
@@ -56,6 +56,25 @@ export const LINKED_FILES = [path.join('backend', '.env')];
...
+ * Linux a symlink reads as a file, so git names the link ITSELF, with no
+ * trailing slash (`node_modules`). Matching only the `node_modules/` prefix
+ * missed that second form and reported the link as uncommitted work — which
+ * made `wt remove` refuse every worktree it had just created.
+ */
+export function isLinkNoise(file) {
+  const f = toPosix(file);
+  if (LINKED_FILES.map(toPosix).includes(f)) return true;
+  return LINKED_DIRS.map(toPosix).some((d) => f === d || f.startsWith(`${d}/`));
+}
```
[the remark this answers is not in the document]
Comment (rimusz) [reply scripts/worktree.mjs:89]: Applied in 46679540 — both lists are now normalised once at module scope (`LINKED_FILES_POSIX` as a `Set`, `LINKED_DIRS_POSIX` as an array), so the function is allocation-free.

I benchmarked it before applying, and the numbers are worth recording because they change where the win actually lands:

| n (status lines) | current | precomputed | saved |
|---|---|---|---|
| 100 | 0.05 ms | 0.13 ms | *below noise* |
| 10,000 | 4.77 ms | 1.71 ms | 3.06 ms (64%) |
| 200,000 | 83.87 ms | 32.43 ms | 51.43 ms (61%) |

`m` is fixed at 3, and in normal use `n` is just the user's changed files — well under a hundred, where the difference is unmeasurable (the n=100 row is noise, not a regression).

Large `n` is only reachable if `.gitignore` stops covering the links and git enumerates all ~30k files beneath them. That is precisely the regression this filter exists to absorb — it deliberately does not trust `.gitignore` to be right, which is why it runs at all. So the saving lands on the backstop path rather than the common one, which seems like the right place for it.

One thing I checked before hoisting: the usual hazard is freezing a value that later changes. Both `LINKED_DIRS` and `LINKED_FILES` are `export const` arrays, and nothing in `scripts/` or `.githooks/` reassigns or mutates either, so there is nothing here that can go stale.

Verified equivalent on all 9 edge cases (including `node_modules.txt` and `node_modules-notes/`), and re-ran the mutation tests against the new shape — the three original mutations are still each killed by a different test, plus two new ones for the precompute's own failure modes:

| Mutation | Killed by |
|---|---|
| restore the trailing slashes | `.gitignore` contract test |
| drop the bare-entry match | POSIX `isLinkNoise` test |
| naive `startsWith(d)` prefix | *never swallows the user's own work* |
| empty the hoisted dirs array | both POSIX **and** Windows spelling tests |
| drop the hard-linked-file check | *matches a hard-linked file* |

18/18 green, full suite 5508 passed, `eol:check` clean.

---

Separately, for anyone reading this thread: the review reports **Files reviewed: 2/3** at **Review effort level: Lite**, so one of the three changed files went unreviewed and the summary does not say which. Flagging it so it is not mistaken for full coverage.
